### PR TITLE
docs: add Stylelint configuration and cleanup

### PR DESCRIPTION
**Pull request by nschonni**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

#### What changes did you make? (Give an overview)

Add Stylelint for SCSS files and address the simpler rules in separate commits. Avoided more of the rules that would have funtional changes like the duplicate rules or vendor prefixes

### CODE REVIEW

File: docs/src/assets/scss/print.scss
Comment (nschonni) [inline docs/src/assets/scss/print.scss:186]: Guessing this might need to be the actual site domain
```suggestion
a[href^="http"]:not([href*="eslint.org"])::after {
```

Comment (nzakas) [reply docs/src/assets/scss/print.scss:186]: Hehe. Definitely seems that way.

**Comment by nzakas** [comment]
@eslint/website-team this is for you folks to decide how to move forward.

**Review by kecrily — APPROVED** [review]
LGTM, I like all the ways to make the code uniform and neat

File: docs/package.json
Comment (harish-sethuraman) [inline docs/package.json:19]:
```suggestion
        "lint:scss": "stylelint \"**/*.scss\""
```

We are trying to unify the script naming across repos. I feel like it'd be better if we go with `lint:scss`
https://github.com/eslint/eslint.org/blob/dae022f4ddd4a31b4fe5c9bb90e650205ab128c4/package.json#L32

**Review by harish-sethuraman — COMMENTED** [review]
Shall we also add a `lint:fix:scss` command too?

**Comment by nschonni** [comment]
> Shall we also add a `lint:fix:scss` command too?

I've seen you use the `lint:fix:*` here. I usually find the plain `lint:*` and `fix:*` more common in other projects (and less to type :smile:)

**Comment by harish-sethuraman** [comment]
I've found `lint:fix` being used in most of the projects that I've worked on 😅  and  [here](https://github.com/eslint/eslint/issues/14827) is an issue regarding standardization in case if you were wondering

**Comment by nschonni** [comment]
> PS: mostly we should avoid manually fixing (running the `lint:fix` commands) lint issues in case they are auto fixable. We can consider adding pre commit hook just in case.

Yeah, I thought about adding the lint-staged for this, but that setup is at the root project, and I figured it was better to keep this dependency inside the `docs/` folder. It might still work, I just didn't dig into getting it setup

File: docs/package.json
Comment (nschonni) [inline docs/package.json:19]:
```suggestion
        "lint:scss": "stylelint \"**/*.scss\"",
        "lint:fix:scss": "npm run lint:scss -- --fix"
```

**Comment by nschonni** [comment]
Feel free to push any suggestions to the branch. I've just been flagging suggestions, but you folks can make the final call on which way you want to go.
I split out all the commits separately, but this can definitely be squash/merged in the end, if you want to accept it.

**Comment by harish-sethuraman** [comment]
We already have some lint-staged scripts running for files inside the docs directory. We can use the same and add stylelint alone? 

https://github.com/eslint/eslint/blob/720ff75beb9f4fdcf2a185fcb8020cf78483fdeb/package.json#L36-L43

**Comment by nzakas** [comment]
We can always add lint-staged later. It may be a bit tricky because it’s at the root of the project and stylelint isn’t installed there. So I’d say if we are good with the other changes that we should merge this and work on lint-staged separately.

**Comment by nzakas** [comment]
@eslint/website-team please review this PR, merge in any suggested changes, and approve so we can move forward. Thanks!

**Comment by harish-sethuraman** [comment]
I'm not able to commit suggested change as I don't have write access to this repo 😅 . Apart from the only `lint:fix:scss` suggestion still to be committed all other changes looks good.

**Review by harish-sethuraman — APPROVED** [review]

File: docs/package.json
Comment (mdjermanovic) [inline docs/package.json:19]:
```suggestion
        "lint:scss": "stylelint \"**/*.scss\"",
        "lint:fix:scss": "npm run lint:scss -- --fix"
```

Copy & paste from https://github.com/eslint/eslint/pull/16379#discussion_r987281984 as GitHub now considers it outdated and doesn't allow to apply the change.

**Comment by mdjermanovic** [comment]
Code blocks look different after this change.

https://eslint.org/docs/head/rules/array-callback-return:

image


https://deploy-preview-16379--docs-eslint.netlify.app/rules/array-callback-return:

image

**Comment by harish-sethuraman** [comment]
I guess some font is missing in the deploy preview?

**Comment by nschonni** [comment]
It seems like the computed CSS is different:
PR:
```
"Inter", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"
```

Live:
```
"Mono Punctuators", "Space Mono", monospace, Consolas, Monaco, "Andale Mono", "Ubuntu Mono", monospace
```

File: docs/.stylelintrc.json
Comment (mdjermanovic) [inline docs/.stylelintrc.json:26]:
```suggestion
        "_site/**"
```

With `_site/`, I'm getting errors in `_site/assets/scss/styles.scss`. I'm guessing this should be a glob that matches files.

File: docs/src/assets/scss/syntax-highlighter.scss
Comment (harish-sethuraman) [inline docs/src/assets/scss/syntax-highlighter.scss:5]:
```suggestion
        var(--mono-font),
        Consolas,
```

Comment (harish-sethuraman) [reply docs/src/assets/scss/syntax-highlighter.scss:5]: I found it 😛  it was a missing comma. Nice find @mdjermanovic

**Comment by nschonni** [comment]
Looks like the missing comma in the font variable was the issue. Preview is looking correct again 
image

Should I rebase to see if there is anything in the changes to `_include/layouts` that might be different from when this was opened?

File: docs/.stylelintrc.json
Comment (harish-sethuraman) [reply docs/.stylelintrc.json:26]: should we be more specific and add only scss and css to ignoreFiles?

**Comment by harish-sethuraman** [comment]
Let's rebase and run lint fix once to avoid miss

**Comment by nschonni** [comment]
Rebased, but left the code review suggestion commits. I can merge those into the particular commit to keep it clean, but I wasn't sure if the plan was for a squash/merge in the end

File: docs/.stylelintrc.json
Comment (mdjermanovic) [reply docs/.stylelintrc.json]: > should we be more specific and add only scss and css to ignoreFiles?

`_site` is the build output, so it makes sense to just ignore the whole directory, as originally intended. But it seems that `ignoreFiles` doesn't allow ignoring directories (perhaps that could be done with `.stylelintignore`).

**Comment by mdjermanovic** [comment]
> Rebased, but left the code review suggestion commits. I can merge those into the particular commit to keep it clean, but I wasn't sure if the plan was for a squash/merge in the end

It's okay to leave all commits, we always squash-merge in the end.

**Comment by nschonni** [comment]
One other one I noticed when I was trying to figure out the Font thing, was that there is some CSS in `<style>` tags in the layout files. Do you want me to add this https://www.npmjs.com/package/stylelint-config-html in this PR, or keep that for a separate thing? It only seems to adjust the intend on the font declaration in `base.html` when I ran it

**Comment by mdjermanovic** [comment]
> Do you want me to add this https://www.npmjs.com/package/stylelint-config-html in this PR, or keep that for a separate thing?

It might be better to evaluate this separately. There are already a lot of changes in this PR, and our `.html` files are not raw HTML so we should doublecheck if linting actually works.

**Review by harish-sethuraman — APPROVED** [review]
LGTM

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
